code/controllers/subsystem/shuttle.dm

```diff
@@ -374,17 +382,34 @@ SUBSYSTEM_DEF(shuttle)
 		priority_announce("Hostile environment detected. \
 			Departure has been postponed indefinitely pending \
 			conflict resolution.", null, 'sound/misc/notice1.ogg', "Priority")
-		for(var/i in hostileEnvironments)
-			if(istype(i, /mob/living/carbon/alien/humanoid/royal/queen) && !hostileEnvTrackPlayed)
-				play_soundtrack_music(/datum/soundtrack_song/bee/mind_crawler, only_station = TRUE)
-				hostileEnvTrackPlayed = TRUE
 	if(!emergencyNoEscape && (emergency.mode == SHUTTLE_STRANDED))
 		emergency.mode = SHUTTLE_DOCKED
 		emergency.setTimer(emergencyDockTime)
 		priority_announce("Hostile environment resolved. \
 			You have 3 minutes to board the Emergency Shuttle.",
 			null, ANNOUNCER_SHUTTLEDOCK, "Priority")
 
+/datum/controller/subsystem/shuttle/proc/checkInfestedEnvironment()
+	for(var/d in infestedEnvironments)
+		var/turf/T = get_turf(d)
+		if(!istype(d) || QDELETED(d) || !is_station_level(T.z)) //If they have been destroyed or left the station Z level, the queen will not trigger this check
```

Comment:
QDELETED(d) requires a static type

Reply:
haha so merging @Penwin0's suggestion is what fucked it